Sphinx version error building ros2_documentation in Dockerfile

@clalancette we don't strictly need this, as it was a way to access documentation without Internet. But FYI this is an error. We can comment out these lines in the Dockerfile.

New error on this line in Dockerfile when building:
https://github.com/osrf/icra2023_ros2_gz_tutorial/blob/2c10f86beb85dbbb5431a68126bf077b34814e39/docker/icra2023_tutorial/Dockerfile#L142-L146

Probably something needs to be updated

```
13.65 Sphinx version error:
13.65 The sphinxcontrib.applehelp extension used by this project needs at least Sphinx v5.0; it therefore cannot be built with this version.
13.88 Traceback (most recent call last):
13.88   File "/home/developer/.local/bin/sphinx-multiversion", line 8, in <module>
13.88     sys.exit(main())
13.88   File "/home/developer/.local/lib/python3.10/site-packages/sphinx_multiversion/main.py", line 338, in main
13.88     subprocess.check_call(cmd, cwd=current_cwd)
13.88   File "/usr/lib/python3.10/subprocess.py", line 369, in check_call
13.88     raise CalledProcessError(retcode, cmd)
13.88 subprocess.CalledProcessError: Command '('/usr/bin/python3', '-R', '-m', 'sphinx', '-D', 'smv_metadata_path=/tmp/tmp8ujzg71l/versions.json', '-D', 'smv_current_version=rolling', '-c', '/home/developer/ros2_documentation', '/tmp/tmp8ujzg71l/913f2b6cfa9f9dd1ae22fc49764b18316204ebe3/source', '/home/developer/ros2_documentation/build/html/rolling')' returned non-zero exit status 2.
13.91 make: *** [Makefile:18: multiversion] Error 1
```

--- a/docker/icra2023_tutorial/Dockerfile
+++ b/docker/icra2023_tutorial/Dockerfile
@@ -140,7 +140,7 @@ RUN colcon mixin add default https://raw.githubusercontent.com/colcon/colcon-mix
 
 # Build and install the ROS 2 documentation
 RUN cd /home/$USERNAME && git clone https://github.com/ros2/ros2_documentation && cd ros2_documentation \
- && pip3 install -r requirements.txt && PATH=$PATH:/home/$USERNAME/.local/bin make multiversion \
+ && pip3 install -r requirements.txt -c constraints.txt && PATH=$PATH:/home/$USERNAME/.local/bin make multiversion \
  && mkdir /home/$USERNAME/docs.ros.org \
  && cp -av /home/$USERNAME/ros2_documentation/build/html/* /home/$USERNAME/docs.ros.org \
  && rm -rf /home/$USERNAME/ros2_documentation

